download/oci_dowload.go

```diff
@@ -0,0 +1,369 @@
+package download
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/base64"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"oras.land/oras-go/pkg/content"
+	"oras.land/oras-go/pkg/oras"
+
+	"github.com/containerd/containerd/remotes/docker"
+	"github.com/open-policy-agent/opa/bundle"
+	"github.com/open-policy-agent/opa/logging"
+	"github.com/open-policy-agent/opa/metrics"
+	"github.com/open-policy-agent/opa/plugins"
+	"github.com/open-policy-agent/opa/plugins/rest"
+	"github.com/open-policy-agent/opa/util"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+
+	"github.com/pkg/errors"
+)
+
+type OCIDownload struct {
+	config             Config                        // downloader configuration for tuning polling and other downloader behaviour
+	client             rest.Client                   // HTTP client to use for bundle downloading
+	path               string                        // path for OCI image as <registry>/<org>/<repo>:<tag>
+	localStorePath     string                        // path for the local OCI storage
+	trigger            chan chan struct{}            // channel to signal downloads when manual triggering is enabled
+	stop               chan chan struct{}            // used to signal plugin to stop running
+	f                  func(context.Context, Update) // callback function invoked when download updates occur
+	sizeLimitBytes     *int64                        // max bundle file size in bytes (passed to reader)
+	bvc                *bundle.VerificationConfig
+	respHdrTimeoutSec  int64
+	wg                 sync.WaitGroup
+	logger             logging.Logger
+	mtx                sync.Mutex
+	stopped            bool
+	persist            bool
+	longPollingEnabled bool
+	store              *content.OCI
+}
+
+// New returns a new Downloader that can be started.
+func NewOCI(config Config, client rest.Client, path, storePath string) DownloaderInterface {
+	//TODO: Extract oci path into configuration
+	localstore, err := content.NewOCI(storePath)
+	if err != nil {
+		panic(err)
+	}
+	return &OCIDownload{
+		config:             config,
+		path:               path,
+		localStorePath:     storePath,
+		client:             client,
+		trigger:            make(chan chan struct{}),
+		stop:               make(chan chan struct{}),
+		logger:             logging.Get(),
```

Comment:
https://pkg.go.dev/github.com/open-policy-agent/opa/logging#Get deprecated -- let's figure out how to get this in as an argument to `New`

Reply:
I used the logger from the client